src/context.cc

```diff
@@ -636,6 +614,40 @@ WasmResult ContextBase::setTimerPeriod(std::chrono::milliseconds period,
   return WasmResult::Ok;
 }
 
+FilterHeadersStatus ContextBase::convertVmCallResultToFilterHeadersStatus(uint64_t result) {
+  if (stop_iteration_) {
+    stop_iteration_ = false;
+    return FilterHeadersStatus::StopIteration;
+  } else if (result > static_cast<uint64_t>(FilterHeadersStatus::StopAllIterationAndWatermark)) {
+    return FilterHeadersStatus::StopAllIterationAndWatermark;
+  }
+  return static_cast<FilterHeadersStatus>(result);
+}
+
+FilterDataStatus ContextBase::convertVmCallResultToFilterDataStatus(uint64_t result) {
+  if (stop_iteration_ || result > static_cast<uint64_t>(FilterDataStatus::StopIterationNoBuffer)) {
+    stop_iteration_ = false;
+    return FilterDataStatus::StopIterationNoBuffer;
+  }
+  return static_cast<FilterDataStatus>(result);
+}
+
+FilterTrailersStatus ContextBase::convertVmCallResultToFilterTrailersStatus(uint64_t result) {
+  if (!stop_iteration_ &&
+      static_cast<FilterTrailersStatus>(result) == FilterTrailersStatus::Continue) {
+    return FilterTrailersStatus::Continue;
+  }
```

Comment:
This is a bit inconsistent with other functions, could you change it to match the style? i.e.
```
if (stop_iteration_ || result > static_cast<uint64_t>(FilterTrailersStatus::StopIteration) {
    stop_iteration_ = false;
    return FilterTrailersStatus::StopIteration;
}
return static_cast<FilterTrailersStatus>(result);
```